src/platform/gpu.h

```diff
@@ -1,11 +1,117 @@
 #pragma once
 
+#include "common/aliases.h"
 #include "platform/native_platform.h"
 
 namespace vx
 {
+struct gpu_buffer;
+struct gpu_texture;
+struct gpu_sampler;
+struct gpu_vertex_desc;
+struct gpu_shader;
+struct gpu_pipeline;
 struct gpu_channel;
 
+enum gpu_vertex_format
+{
+    gpu_vertex_format_float,
+    gpu_vertex_format_float2,
+    gpu_vertex_format_float3,
+    gpu_vertex_format_float4,
+
+    gpu_vertex_format_rgba8_unorm,
+};
+
+enum gpu_index_type
+{
+    gpu_index_type_u16,
+    gpu_index_type_u32,
+};
+
+enum gpu_pixel_format
+{
+    gpu_pixel_format_rgba8_unorm,
+};
+
+enum gpu_filter_mode
+{
+    gpu_filter_mode_nearest,
+    gpu_filter_mode_linear,
+};
+
+enum gpu_shader_type
+{
+    gpu_shader_type_vertex,
+    gpu_shader_type_fragment,
+};
+
+enum gpu_primitive_type
+{
+    gpu_primitive_type_point,
+    gpu_primitive_type_line,
+    gpu_primitive_type_line_strip,
+    gpu_primitive_type_triangle,
+    gpu_primitive_type_triangle_strip,
+};
+
+struct gpu_scissor_rect
+{
+    u32 x, y, w, h;
+};
+
+struct gpu_viewport
+{
+    float x, y, w, h, znear, zfar;
+};
+
+gpu_buffer* gpu_buffer_create(gpu_device* gpu, usize size);
```

Comment:
How does returning pointers to structs mesh with a possible opengl backend? I would imagine that if we just had a `gpu_buffer_t` (and others) declared as a `u64`, we could stuff a pointer into it, or stick an opengl buffer object and target into it.